### ISSUE TITLE
Add option to disable/hide Quarto panel

### ISSUE THREAD

**Issue by cbrnr** [quarto-dev/quarto-vscode#21]
I don't really need the Quarto panel, which always appears in the lower panel (even when no Quarto document is open). It would be great if (1) the panel is shown only when Quarto files are open and/or (2) there was an option to disable/hide the panel completely.

**Comment by jjallaire** [comment]
Unfortunately I haven't yet figured out a way to make the quarto panel only appear when used (it's registered within the extension's manifest and consequently always appears).

**Comment by mcanouil** [comment]
Hum, you can hide the panel, isn't it enough?

https://user-images.githubusercontent.com/8896044/169410132-437e9cf4-2e81-4cc3-a7d7-6262484a9dc2.mov


Although, it would be great if quarto rendering occurred in "output" rather than in terminal panel, i.e., like for vscode-r and RMarkdown document.
<img width="1450" alt="image" src="https://user-images.githubusercontent.com/8896044/169410483-0dc42912-b566-45a5-950c-837bb0627d37.png">

**Comment by jjallaire** [comment]
We have gone back and forth between the terminal and the output pane (most recently switching back to the terminal here: https://github.com/quarto-dev/quarto-vscode/commit/df686596c2bc9fecc6bb4894e2f1d3a44fb0c93f). The issue with the output pane is that Quarto Preview is a long-running process (which does re-renders in place for higher performance) and users often want/need to interrupt it (which has some UX challenges as there is no way to add a stop button to "output"). At some point we may make this a user option since we have the code available to implement both.

**Comment by cbrnr** [comment]
Thanks @mcanouil, I didn't know that you could hide panels like this. Of course this is good enough. However, UX-wise it would be even better if it was only there when Quarto files are open. The Julia and R extensions behave like this. Even though I have installed them, they only show up once I open a relevant file.

**Comment by mcanouil** [comment]
@cbrnr vscode-r does not include a panel, but an item in the sidebar.
If this is what you meant, contextualise help from the sidebar would make it useful in my case and maybe allow a more dynamic prompt of this (or stick at the hide/show option as for the panel).
-> I usually have a terminal open, the sidebar, editor and preview opened, thus so far I never used Quarto panel because the focus is on the terminal.

**Comment by cbrnr** [comment]
I'm not sure what exactly I want, but I don't like extensions to clutter my panels/sidebar/... At least for the sidebar, most extensions only appear there when some relevant file is open. Same for the status bar, e.g. the Python version is only shown when I have a Python file open (and therefore the Python extension is activated). Not sure about the Quarto panel, I have never seen an extension add something there. I'd prefer consistency, so if most extensions provide a sidebar panel, then this could be the way to go. The Quarto panel is always empty for me, so I really don't need it.

**Comment by jjallaire** [comment]
I agree 100% that the Quarto panel should only show after you've e.g. opened a qmd file (or ideally _only_ when a qmd file is active in the editor). So far though I have not found an API that lets me programmatically hide/dispose the panel (will keep looking though)

**Comment by mcanouil** [comment]
What about the use of the sidebar (instead of panel) as from vscode-r ?
The item (thus the vscode-r extension) only appears if R is installed, maybe possible to add a condition on open/active editor.

**Comment by jjallaire** [comment]
Okay, I've just changed the "assist panel" to:

1) Appear in the sidebar rather than the panel (it's a sub-view of "Explorer" and can actually be moved to the panel or right sidebar by the user if preferred)

2) Only appear on demand (it won't appear even if the Quarto extension is loaded unless you execute the **Quarto: Show Assist Panel** command or **Quarto: Math Preview** command)

This is available in v1.21.0 which is published to the marketplace now.

**Comment by mcanouil** [comment]
Awesome!

**Comment by cbrnr** [comment]
I agree that manually hiding the panel is OK, but it is possible to only show the panel when the extension is active, see e.g. https://github.com/julia-vscode/julia-vscode/pull/2998/files.

This would be the ideal solution IMO.
